openmeter/meter/validate.go

```diff
@@ -0,0 +1,68 @@
+package meter
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/oliveagle/jsonpath"
+)
+
+// ValidateEvent validates an event against a meter.
+func ValidateEvent(meter Meter, ev event.Event) error {
+	// Parse CloudEvents data
+	var data interface{}
+
+	err := ev.DataAs(&data)
+	if err != nil {
+		return errors.New("cannot unmarshal event data")
+	}
+
+	// We can skip count events as they don't have value property
+	if meter.Aggregation == MeterAggregationCount {
+		return nil
+	}
+
+	// Get value from event data by value property
+	rawValue, err := jsonpath.JsonPathLookup(data, meter.ValueProperty)
+	if err != nil {
+		return fmt.Errorf("event data is missing value property at %q", meter.ValueProperty)
+	}
+
+	if rawValue == nil {
+		return errors.New("event data value cannot be null")
+	}
+
+	// Aggregation specific value validation
+	switch meter.Aggregation {
+	// UNIQUE_COUNT aggregation requires string property value
+	case MeterAggregationUniqueCount:
+		switch rawValue.(type) {
+		case string, float64:
```

Comment:
`float64` is also acceptable? Based on the comment, it shouldn't be.

Reply:
That part of the code is just moved somewhere else, it's not actually new. I'm not sure what's the correct behavior.